woocommerce/woocommerce-gutenberg-products-block
# Add descriptions for all WC templates and refactor Search template hierarchy loading logic

This PR updates the way template **titles** and **descriptions** are being loaded, by mirroring [how WP core is doing it](https://github.com/WordPress/WordPress/blob/d9266d5bfd6aed72907414728d83f1053c1c2a7a/wp-includes/block-template-utils.php#L114), moving the relevant code to the `BlockTemplateUtils.php` file. In addition, it moves Search template hierarchy loading logic to `BlockTemplatesController.php`.

Fixes #6233 

### Screenshots
|Before|After|
|-|-|
|![bifor_fixed](https://user-images.githubusercontent.com/905781/165815026-408dccff-ea16-4766-8a65-9696866e6f92.jpg)|![after_-_fixed](https://user-images.githubusercontent.com/905781/165815040-723bb981-5cc2-4787-a38d-d9dee3e12757.jpg)|

### Manual Testing

How to test the changes in this Pull Request:

**Testing template descriptions**
1. Activate a **block** theme, like Twenty Twenty Two
2. Open the **Appearance > Editor (Beta)**
3. Using the dropdown arrow next to the Template name, select the **Browse all templates** button
4. Confirm that all WooCommerce templates have descriptions

**Testing the search template**
1. Activate a **block** theme, like Twenty Twenty Two
2. Make sure there's at least 1 product added
3. Run a product search: `?s={keyword}&post_type=product` and make sure the products are being displayed correctly in a grid (using the search template, instead of the default one).

### User Facing Testing
These are steps for user testing (where "user" is someone interacting with this change that is not editing any code).

### Changelog

> Add descriptions to the FSE WooCommerce Templates in the Editor UI
